iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/helplets.inl

```diff
@@ -22,6 +22,40 @@ namespace iox
 {
 namespace cxx
 {
+template <uint64_t StringCapacity>
+inline bool isValidPathEntry(const string<StringCapacity>& name) noexcept
+{
+    const string<StringCapacity> currentDirectory(".");
+    const string<StringCapacity> parentDirectory("..");
+
+    if (name == currentDirectory || name == parentDirectory)
+    {
+        return true;
+    }
+
+    const auto nameSize = name.size();
+
+    for (uint64_t i = 0; i < nameSize; ++i)
+    {
+        const char c = name.c_str()[i];
+        if (!((internal::ASCII_A <= c && c <= internal::ASCII_Z)
+              || (internal::ASCII_CAPITAL_A <= c && c <= internal::ASCII_CAPITAL_Z)
+              || (internal::ASCII_0 <= c && c <= internal::ASCII_9) || c == internal::ASCII_MINUS
+              || c == internal::ASCII_DOT || c == internal::ASCII_COLON || c == internal::ASCII_UNDERSCORE))
```

Comment:
How about `+%<>[](){}~#`? Would we consider some of them valid character?

Reply:
For the time being no. If they are valid characters they would have to fulfill the following requirements:

* Valid characters for the file systems:
    - linux: ext3, ext4, btrfs
    - windows: ntfs, exfat, fat
    - freebsd: ufs, ffs
    - apple: apfs
    - qnx: etfs?
    - android: ext3, fat (since we would like to support it in the future)

* The characters must be also supported by the corresponding API. The windows API does not support for instance dots at the end. Who knows why?!

So to conclude this. I would assume that we can add more characters to the list but in the first approach I just wanted to make sure that we have a meaningful minimum viable set of characters available which we can extend later.

And for that we would have to write unit tests by actually creating those files on the different file systems and access those files with the native APIs (most likely only windows). This can be ugly ...